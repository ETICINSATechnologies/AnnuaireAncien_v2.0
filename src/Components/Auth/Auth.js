class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
