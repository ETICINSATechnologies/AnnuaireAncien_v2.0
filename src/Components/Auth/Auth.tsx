class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
