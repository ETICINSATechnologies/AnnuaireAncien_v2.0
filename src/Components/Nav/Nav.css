.Nav {
    display: grid;
    grid-template-columns: 1fr 3vw;

    border-bottom: 0.3vmin solid black;
    width: 100%;
}

.Nav .Menu {
    grid-column: 1 / 3;
    grid-row: 1;
    display: flex;
    justify-content: center;

    width: 100%;
}

.Nav .button {
    margin: auto 0.8vmax;
    vertical-align: middle;
    padding: 0;
    background-color: Transparent;
    border: none;

    color: black;
    text-decoration: none;
    font-size: 2vmax;
    white-space: nowrap;

}

.Nav .button:hover {
    color: rgba(0, 158, 17, 1);
    cursor: pointer;
}

.Nav .help_link {
    display: flex;
    grid-column: 2;
    grid-row: 1;
    vertical-align: middle;
    justify-content: right;
    width: 100%;
    height: 100%;
}

.Nav .help_icon {
    width: 1.8vmax;
    height: 1.8vmax;
    vertical-align: middle;
    margin: auto;
}