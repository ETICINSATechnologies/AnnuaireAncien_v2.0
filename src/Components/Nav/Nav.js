import React, {Component} from 'react';
import './Nav.css';

import help_icon from '../../Images/help_icon.png'
import Auth from "../Auth/Auth";

class Nav extends Component {
    constructor(props) {
        super(props);
        this.state = {
            connection: {
                href: '/connection',
                value: 'se connecter'
            },
            disconnection: {
                href: '/',
                value: 'se d√©connecter'
            },
            home: {
                href: '/',
                value: 'accueil'
            },
            search: {
                href: '/search',
                value: 'rechercher'
            },
            profile: {
                href: '/profile',
                value: 'mon profil'
            }
        }
    }

    render() {
        let navButtons = [];
        if (this.props.buttons){
            for (let i = 0; i < this.props.buttons.length; i++) {
                let currentButton = this.state[this.props.buttons[i]];
                if (this.props.buttons[i] === 'disconnection')
                    navButtons.push(
                        <a className="button" key={i} href={currentButton.href} onClick={Auth.disconnect}>
                            {currentButton.value}
                        </a>
                    );
                else
                    navButtons.push(
                        <a className="button" key={i} href={currentButton.href}>
                            {currentButton.value}
                        </a>
                    );
            }
        }

        return (
            <nav className='Nav'>
                <div className='Menu'>
                    {navButtons}
                </div>
                <a className="help_link" href="/help">
                    <img className="help_icon" src={help_icon} alt=""/>
                </a>
            </nav>

        );
    }
}

export default Nav;
