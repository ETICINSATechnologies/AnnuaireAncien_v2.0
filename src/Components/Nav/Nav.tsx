import React from 'react';
import {Link} from "react-router-dom";

import './Nav.css';
import helpIcon from '../../Images/help_icon.png';
import Auth from "../Auth/Auth";


type NavProps = {
    buttons: string[]
}

type redirect = {
    href: string
    value: string
}

const navRedirect = {
    connection: {
        href: '/connection',
        value: 'se connecter'
    } as redirect,
    disconnection: {
        href: '/',
        value: 'se d√©connecter'
    } as redirect,
    home: {
        href: '/',
        value: 'accueil'
    } as redirect,
    search: {
        href: '/search',
        value: 'rechercher'
    } as redirect,
    profile: {
        href: '/profile',
        value: 'mon profil'
    } as redirect,
};

const Nav: React.SFC<NavProps> = ({buttons}: NavProps) => {
    let navButtons = buttons.map((button: string, i: number) => {
        return (
            <Link className="button" key={i} to={(navRedirect as any)[button].href}
                  onClick={button === 'disconnection' ? Auth.disconnect : undefined}>
                {(navRedirect as any)[button].value}
            </Link>
        );
    });

    return (
        <nav className='Nav'>
            <div className='Menu'>{navButtons}</div>
            <Link className="help_link" to="/help">
                <img className="help_icon" src={helpIcon} alt=""/>
            </Link>
        </nav>

    );
};

export default Nav;
