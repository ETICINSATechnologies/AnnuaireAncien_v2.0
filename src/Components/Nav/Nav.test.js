import React from 'react';
import Nav from './Nav';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Auth from "../Auth/Auth";



Enzyme.configure({adapter: new Adapter()});

describe( 'Nav component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<Nav />).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('Nav unit test', () => {
        expect(Enzyme.shallow(<Nav/>)
            .find('nav')
            .length
        ).toBe(1)
    });
    
});