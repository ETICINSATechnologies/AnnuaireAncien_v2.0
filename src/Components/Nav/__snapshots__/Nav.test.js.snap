// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders without crashing 1`] = `
<nav
  className="Nav"
>
  <div
    className="Menu"
  />
  <a
    className="help_link"
    href="/help"
  >
    <img
      alt=""
      className="help_icon"
      src="help_icon.png"
    />
  </a>
</nav>
`;
